aiidalab_widgets_base/structures.py

```diff
@@ -215,143 +241,160 @@ def get_example_structures(examples):
 
     def _on_select_structure(self, change):  # pylint: disable=unused-argument
         """When structure is selected."""
-        if not self._select_structure.value:
-            return
-        structure_ase = get_ase_from_file(self._select_structure.value)
-        self.on_structure_selection(structure_ase=structure_ase, name=self._select_structure.label)
+
+        self.structure = None if not self._select_structure.value else get_ase_from_file(self._select_structure.value)
+
+    @default('structure')
+    def _default_structure(self):
+        return None
 
 
 class StructureBrowserWidget(ipw.VBox):
     """Class to query for structures stored in the AiiDA database."""
+    structure = Union([Instance(Atoms), Instance(Data)], allow_none=True)
 
     def __init__(self):
-        # Find all process labels
-        qbuilder = QueryBuilder()
-        qbuilder.append(WorkChainNode, project="label")
-        qbuilder.order_by({WorkChainNode: {'ctime': 'desc'}})
-        process_labels = {i[0] for i in qbuilder.all() if i[0]}
-
-        layout = ipw.Layout(width="900px")
-        self.mode = ipw.RadioButtons(options=['all', 'uploaded', 'edited', 'calculated'],
-                                     layout=ipw.Layout(width="25%"))
-
-        # Date range
-        self.dt_now = datetime.datetime.now()
-        self.dt_end = self.dt_now - datetime.timedelta(days=10)
-        self.date_start = ipw.Text(value='', description='From: ', style={'description_width': '120px'})
-
-        self.date_end = ipw.Text(value='', description='To: ')
-        self.date_text = ipw.HTML(value='<p>Select the date range:</p>')
-        self.btn_date = ipw.Button(description='Search', layout={'margin': '1em 0 0 0'})
-        self.age_selection = ipw.VBox(
-            [self.date_text, ipw.HBox([self.date_start, self.date_end]), self.btn_date],
-            layout={
-                'border': '1px solid #fafafa',
-                'padding': '1em'
-            })
 
-        # Labels
-        self.drop_label = ipw.Dropdown(options=({'All'}.union(process_labels)),
+        # Structure objects we want to query for.
+        self.what_query_for = (DataFactory('structure'), DataFactory('cif'))
```

Comment:
I propose to rename this attribute to "query_structure_type" or so. The current name is not very meaningful.

Reply:
Thanks for the suggestion. I could not come up with a good name here.

Reply:
done